src/JuMP.jl

```diff
@@ -100,6 +100,9 @@ mutable struct Model <: AbstractModel
     # obj#::QuadExpr
     # objSense::Symbol
 
+    # We use the constraint reference value to have a concrete type
+    constrainttosolverconstraint::Dict{MOICON,MOICON}
```

Comment:
Doesn't this field need to be initialized in the Model() constructor? Why doesn't this code crash?

Reply:
I am doing the same than what you did with `variabletosolvervariable`. It is initialized in every call of `attach`. If the variable is accessed before `attach` is ever called, there will be an `Undef...` error but this does not happen since the dictionary only have sense once a solver is attached.